config/initializers/filter_parameters.rb

```diff
@@ -0,0 +1,46 @@
+# Be sure to restart your server when you modify this file.
+
+# Filter sensitive data in JSON parameters from logs
+#
+# JSON parameters are in the key, they look something like this:
+# Parameters: {"{\"email\":null,\"username\":\"karen\",\"password\":\"asdfasdf\"}"=>[FILTERED]}
+#
+# The way the filter parameters work is that first, all the regexp and string
+# filters are matched against the key, if it matches, the value is filtered and
+# the other filters (proc filters) are NOT run.
+#
+# See actionpack-3.2.17/lib/action_dispatch/http/parameter_filter.rb
+#
+# So if we have a string filter "password", the JSON parameter that contains
+# "password" will have the value filtered, but the actual sensitive data
+# (asdfasdf in the example) is still in the log.
+#
+# The way to get around this is to remove the string filters and include them
+# in the filter proc below.
+
+# First get all the filter parameters
+filter_parameters = Rails.application.config.filter_parameters
+
+# Get all the string filter parameters (these are used to match against the key
+# of the parameters and filter the value
+string_filters = filter_parameters.select { |p| p.is_a?(Symbol) || p.is_a?(String) }
```

Comment:
Is it possible that since `string_filters` is removed from the overall filters, is it possible we're removing other non-password filters here?